appengine/flexible/endpoints/clients/service_to_service_google_id_token/main.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0(the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http:  // www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Example of calling a Google Cloud Endpoint API from Google App Engine
+Default Service Account using Google ID token."""
+
+import base64
+import httplib
+import json
+import time
+import urllib
+
+from google.appengine.api import app_identity
+import webapp2
+
+DEFAUTL_SERVICE_ACCOUNT = "YOUR-CLIENT-PROJECT-ID@appspot.gserviceaccount.com"
+HOST = "YOUR-SERVER-PROJECT-ID.appspot.com"
+TARGET_AUD = "YOUR-SERVER-PROJECT-ID@appspot.gserviceaccount.com"
+
+
+def generate_jwt():
+    """Generates a signed JSON Web Token using the Google App Engine default
+    service account."""
+    now = int(time.time())
+
+    header_json = json.dumps({
+        "typ": "JWT",
+        "alg": "RS256"})
+
+    payload_json = json.dumps({
+        "iat": now,
+        # expires after one hour.
+        "exp": now + 3600,
+        # iss is the Google App Engine default service account email.
+        "iss": DEFAUTL_SERVICE_ACCOUNT,
+        # scope must match 'audience' for google_id_token in the security
+        # configuration in your swagger spec.
+        "scope": TARGET_AUD,
+        # aud must be Google token endpoints URL.
+        "aud": "https://www.googleapis.com/oauth2/v4/token"
+    })
+
+    headerAndPayload = '{}.{}'.format(base64.urlsafe_b64encode(header_json),
```

Comment:
nit: break after the opening `(` so that this doesn't hang, e.g.:

```python
headerAndPayload = '{}.{}'.format(
    base64.urlsafe_b64encode(header_json),
    base64.urlsafe_b64encode(payload_json))
```